### PR TITLE
docs: fix, build and persist javadocs

### PR DESCRIPTION
**Pull request by lionel-nj**
**Summary:**

This PR fixes, and adds the process to build and persist javadoc as an artefact in the GitHub workflow.

**At present**
When running `./gradlew aggregateJavadocs` command from the Terminal, we get a list of errors, and javadocs can not be aggregated. 
Also, the `doc` task is not longer included as part of GitHub Actions.

Capture d’écran, le 2021-02-10 à 17 35 54

**Expected behavior:** 

Javadoc should be generated and persisted in the artefact list.

1️⃣ Javadocs is aggregated when running `./gradlew aggregateJavadocs` command from the Terminal 

Capture d’écran, le 2021-02-10 à 17 36 38

2️⃣ `doc` task is executed as par of GitHub workflow
Capture d’écran, le 2021-02-03 à 16 18 08

3️⃣ Javadoc is persisted as an artifact of GtHub Actions as https://github.com/MobilityData/gtfs-validator/actions/runs/535012922 demonstrates
Capture d’écran, le 2021-02-03 à 16 17 06

✅ Javadocs seems to render properly:

Capture d’écran, le 2021-02-10 à 17 47 03

renders as :

Capture d’écran, le 2021-02-10 à 17 46 52

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: -new feature short description-" (PR title must follow the [conventional commit specification](https://www.conventionalcommits.org/en/v1.0.0/))
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: core/src/main/java/org/mobilitydata/gtfsvalidator/annotation/Required.java [hunk 1]
```diff
@@ -27,14 +27,13 @@
...
  *   public interface GtfsAgencySchema extends GtfsEntity {
- *       @FieldType(FieldTypeEnum.ID)
- *       @PrimaryKey
+ *       {@literal @}FieldType(FieldTypeEnum.ID)
+ *       {@literal @}PrimaryKey
  *       String agencyId();
  *
- *       @Required
- *       String agencyName();
+ *       {@literal @}String agencyName();
```
Comment (maximearmstrong) [inline core/src/main/java/org/mobilitydata/gtfsvalidator/annotation/Required.java:36]: Is "Required" removed willingly?

**Review by maximearmstrong — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/mobilitydata/gtfsvalidator/annotation/Required.java (hunk 1 above)
Comment (lionel-nj) [reply core/src/main/java/org/mobilitydata/gtfsvalidator/annotation/Required.java]: Not at all, thanks for flagging that!

**Review by maximearmstrong — APPROVED** [review]
LGTM

File: core/src/main/java/org/mobilitydata/gtfsvalidator/annotation/PrimaryKey.java [hunk 2]
```diff
@@ -32,20 +32,20 @@
...
  *       String stopId();
  *   }
  * </pre>
  *
  * <p>This also adds a method to the container class to find an entity by its primary key.
  *
  * <pre>
- *   @Generated
- *   public class GtfsStopTableContainer extends GtfsTableContainer<GtfsStop> {
+ *   {@literal @}Generated
```
Comment (barbeau) [inline core/src/main/java/org/mobilitydata/gtfsvalidator/annotation/PrimaryKey.java:47]: @aababilov Do any of these Javadoc changes cause issues for you at Google? 

As @lionel-nj states in https://github.com/MobilityData/gtfs-validator/pull/718#issue-567114622 these are required to get the Javadocs to generate and render correctly. @lionel-nj looked at the Google Java Style Guide and it doesn't seem to cover this type of formatting, and as https://reflectoring.io/howto-format-code-snippets-in-javadoc/ indicates there are different ways to solve this problem. It looks like the changes @lionel-nj made here resolve the issue and can be merged, providing they aren't problematic for you.

**Review by barbeau — APPROVED** [review]
LGTM, although prior to merging I'd like to get @aababilov's feedback as mentioned in-line, given he wrote this originally.

**Pull request merged**